### PR TITLE
LIHADOOP-28747 Hadoop plugin's test plugin should have functionality …

### PR DESCRIPTION
**Pull request by nntnag17**
…to add assertions on tests.

### CODE REVIEW

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import com.linkedin.gradle.hadoopdsl.HadoopDslMethod;
+import com.linkedin.gradle.hadoopdsl.NamedScope;
+import com.linkedin.gradle.hadoopdsl.Workflow;
+import org.gradle.api.Project;
+
+
+/**
+ * The AssertExtension allows users to add assertions in their workflow tests
+ */
+class AssertExtension extends Workflow {
```
Comment (pranayhasan) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy]: As this is a workflow, lets rename to AssertWorkflow / AssertionWorkflow.

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  List<String> asserts;
+
+  /**
+   * Base constructor for a AssertExtension
+   *
+   * @param name The assertion name
+   * @param project The Gradle project
+   */
+  AssertExtension(String name, Project project) {
```
Comment (pranayhasan) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy]: AssertionWorkflowExtension?

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.TaskAction;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+/**
+ * AzkabanBlockedFlowStatusTask fetches the status of the flows when they are completed.*/
```
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy]: Done

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy [hunk 4]
```diff
@@ -593,4 +593,22 @@ class AzkabanHelper {
     }
   }
 
+  /**
+   * Prints a spinning loader
+   * @param times The number of times spin should occur
+   * @param delay The delay in each spin
+   */
+  static void printSpinningLoader(long times, long delay) {
```
Comment (pranayhasan) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy]: Can we move the utility functions like these which are being used outside hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban from AzkabanHelper to a new helper class in utils?

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+ * @param sessionId
+ */
+  void getBlockedAzkabanFlowStatus(String sessionId) {
+
+    // list of all the flows defined in the project
+    List<String> flows = getSortedFlows(sessionId);
+
+
+    // list of all the recent execution ids of the flows defined in the project
+    List<String> execIds = getRecentFlowExecutionIds(sessionId, flows);
```
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy]: I will take this up as a separate PR. I'm planning to do some refactoring in the previous couple of PRs that I've sent. I'll refactor it as part of those changes.

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+ * the Azkabban server and outputs the job that finished immediately.
+ * @param sessionId
+ */
+  void getBlockedAzkabanFlowStatus(String sessionId) {
+
+    // list of all the flows defined in the project
+    List<String> flows = getSortedFlows(sessionId);
+
+
+    // list of all the recent execution ids of the flows defined in the project
```
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy]: done

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+ * getBlockedAzkabanFlowStatus waits for the flow to finish. It polls
+ * the Azkabban server and outputs the job that finished immediately.
+ * @param sessionId
+ */
+  void getBlockedAzkabanFlowStatus(String sessionId) {
+
+    // list of all the flows defined in the project
+    List<String> flows = getSortedFlows(sessionId);
+
+
```
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy]: done

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy (hunk 4 above)
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy]: Done

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy (hunk 2 above)
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy]: Done

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy (hunk 1 above)
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/AssertExtension.groovy]: Done

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanFlowStatusTask.groovy [hunk 8]
```diff
@@ -90,8 +100,26 @@ class AzkabanFlowStatusTask extends DefaultTask {
         }
       }
     }
+    return execIds;
+  }
+
+  /**
+   * Uploads the zip file to Azkaban.
```
Comment (nntnag17) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanFlowStatusTask.groovy]: Done

**Pull request merged**